### PR TITLE
REPL meeting minutes.

### LINKED ISSUE #213: Modifications to rustc for the Rust REPL
As several of you know, I managed to successfully create a proof-of-concept Rust REPL about a couple of months ago. This necessitated (at least, given a clean and sane approach) a few — essentially minor — changes to rustc.

As discussed with @nikomatsakis and @Mark-Simulacrum, the we thought the best way to proceed with this would be to create a design document that summarises my approach to creating the REPL, along with some technical details. The idea is to get some feedback either before or during the FCP process, from which I can proceed by submitting PRs.

[**HackMD Document**](https://hackmd.io/GJokfI0wQ0i4SIgRbFTmfw)

See the [first REPL-related PR](https://github.com/rust-lang/rust/pull/64648) (currently pending this issue and FCP) for more examples of the sorts of changes I am making (although that will need to be updated somewhat).

### PR DESCRIPTION
**Pull request by pnkfelix**
Closes #213

### CODE REVIEW

File: content/minutes/design-meeting/2019-11-29-repl-meeting.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+# Big picture overview of design, and on-the-fly questions about it
+
+- Read (to an AST), Compile (AST to MIR), Eval (the MIR to a value), Print (the value). and Loop.
+- Question: Is this going to be a tool analogous to miri, where check-in's that break it may not break CI immediately, but rather will just file follow-up bugs?
+  - Answer 1: it has to be a tool; it depends on miri (or a fork of miri)
+  - Answer 2: it will be even less than a submodule (where a CI automatically files follow-up bugs): There
+    won't be any automatic filing of follow-up bugs (alexreg has said
+    they will handle maintenance manually); this is how miri and
+    clippy had initially started. It could become a swubmodule later.
```
Comment (oli-obk) [inline content/minutes/design-meeting/2019-11-29-repl-meeting.md]:
```suggestion
    clippy had initially started. It could become a submodule later.
```

Comment (pnkfelix) [reply content/minutes/design-meeting/2019-11-29-repl-meeting.md]: (I just do transcriptions on behalf of elmer fudd)

File: content/minutes/design-meeting/2019-11-29-repl-meeting.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+  - no objections to these additions
+
+- "Made stack pop behaviour more flexible, so as to allow the cleanup flag to be independent of wherever the action is null or a “goto”."
+  - Question: What are the stack pop changes?
+  - Answer: it's important that we can exit a block / pop off the stack but still not perform cleanup (so as to be able to serialise the values of the locals when execution is done).
+  - Question: Is the cleanup the running of destructors?
+  - Answer: more like clearing the local's memory
+  - no objections to this addition
+
+# Any remainning conerns from stake holders?
```
Comment (bjorn3) [inline content/minutes/design-meeting/2019-11-29-repl-meeting.md]: *remaining

**Comment by alexreg** [comment]
Thanks a lot, @pnkfelix. This summarises everything nicely, I feel.

**Comment by thor314** [comment]
Sorry for pestering, but I haven't seen much about a REPL in 3 months, where did this land, slash, is there a REPL being built?

**Comment by adamency** [comment]
for posterity's sake, the discussion has continued here: https://github.com/rust-lang/rfcs/issues/655, and it is still open and ongoing. No proper official REPL for Rust is available as of today.

**Pull request merged**